cranelift-codegen/meta/src/isa/x86/legalize.rs

```diff
@@ -82,6 +85,15 @@ pub fn define(shared: &mut SharedDefinitions, x86_instructions: &InstructionGrou
         vec![def!((res_lo, res_hi) = x86_smulx(x, y))],
     );
 
+    // SIMD
+//    group.legalize(
+//        def!(y = splat(x)),
+//        vec![
+//            def!(a = bitcast(x)),
+//            def!(y = x86_pshuf(a))
+//        ]
+//    );
```

Comment:
This is a sticking point for me: if I uncomment this the code generation will fail:
```shell
$ CRANELIFT_VERBOSE=1 cargo build && target/debug/clif-util test -v filetests/isa/x86/simd.clif 
   Compiling cranelift-codegen-meta v0.30.0 (/home/abrown/Code/cranelift/cranelift-codegen/meta)
   Compiling cranelift-codegen v0.30.0 (/home/abrown/Code/cranelift/cranelift-codegen)
error: failed to run custom build command for `cranelift-codegen v0.30.0 (/home/abrown/Code/cranelift/cranelift-codegen)`

Caused by:
  process didn't exit successfully: `/home/abrown/Code/cranelift/target/debug/build/cranelift-codegen-1f59b278cc90c54e/build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-changed=/home/abrown/Code/cranelift/cranelift-codegen/build.rs

--- stderr
thread 'main' panicked at 'assertion failed: vars_tv.contains(arg)', cranelift-codegen/meta/src/cdsl/type_inference.rs:367:17
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

Reply:
Note that the variables `a`, `x`, and `y` refer to variables that might be typed with a type variable; since the build error happens during type inference, that might be it. The error message is very cryptic and could probably be enhanced in any case. I am going to try this locally.